packages/next-on-pages/templates/cache/adaptor.ts

```diff
@@ -87,7 +87,7 @@ export class CacheAdaptor {
 				await this.loadTagsManifest();
 
 				// Check if the cache entry is stale or fresh based on the tags.
-				const tags = getDerivedTags(data.value.data.tags ?? []);
+				const tags = getDerivedTags(data.value.tags ?? []);
```

Comment:
```suggestion
				const tags = getDerivedTags(
					data.value.tags ?? data.value.data.tags ?? []
				);
```

This should also check for tags in `data.value.data.tags` otherwise it won't work with existing cache entries.